tests/core/data/test_data_module.py

```diff
@@ -139,6 +140,19 @@ def on_fit_end(self) -> None:
     trainer.test(model, datamodule=dm)
     trainer.predict(model, datamodule=dm)
 
+    # Test that plain lightning module works with FlashDataModule
+    class SampleBoringModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.layer = torch.nn.Linear(2, 1)
+
+    model = SampleBoringModel()
+    trainer = Trainer(fast_dev_run=True)
+    trainer.fit(model, datamodule=dm)
+    trainer.validate(model, datamodule=dm)
+    trainer.test(model, datamodule=dm)
+    trainer.predict(model, datamodule=dm)
+
```

Comment:
I think this test file in its current state, can be updated. There is a lot that can be better-engineered, but let's keep it for another PR! :)